src/lib.rs

```diff
@@ -1,4 +1,6 @@
 #![no_std]
+#![feature(generic_associated_types)]
+#![feature(type_alias_impl_trait)]
```

Comment:
There's not really a good way to do async on embedded without using the nightly compiler features. It would be good to restrict these to async only if possible, though not sure how to do that.

Reply:
I hoped on to the Rust Discord and they said this should work:
```rust
#![cfg_attr(feature = "async", feature(generic_associated_types))]
#![cfg_attr(feature = "async", feature(type_alias_impl_trait))]
```